src/main/java/uk/gov/justice/digital/job/DataHubJob.java

```diff
@@ -90,18 +98,21 @@ private void batchProcessor(JavaRDD<byte[]> batch) {
             val rowRdd = batch.map(d -> RowFactory.create(new String(d, StandardCharsets.UTF_8)));
             val dataFrame = converter.convert(rowRdd);
 
-            getTablesInBatch(dataFrame).forEach(table -> {
+            getTablesInBatch(dataFrame).forEach(tableInfo -> {
                 try {
-                    val dataFrameForTable = extractDataFrameForSourceTable(dataFrame, table);
+                    val dataFrameForTable = extractDataFrameForSourceTable(dataFrame, tableInfo);
                     dataFrameForTable.persist();
 
-                    rawZone.process(spark, dataFrameForTable, table);
+                    rawZone.process(spark, dataFrameForTable, tableInfo);
 
-                    val structuredLoadDataFrame = structuredZoneLoad.process(spark, dataFrameForTable, table);
-                    curatedZoneLoad.process(spark, structuredLoadDataFrame, table);
+                    val structuredLoadDataFrame = structuredZoneLoad.process(spark, dataFrameForTable, tableInfo);
+                    curatedZoneLoad.process(spark, structuredLoadDataFrame, tableInfo);
 
-                    val structuredIncrementalDataFrame = structuredZoneCDC.process(spark, dataFrameForTable, table);
-                    curatedZoneCDC.process(spark, structuredIncrementalDataFrame, table);
+                    val structuredIncrementalDataFrame = structuredZoneCDC.process(spark, dataFrameForTable, tableInfo);
+                    val curatedCdcDataFrame = curatedZoneCDC.process(spark, structuredIncrementalDataFrame, tableInfo);
+
+                    if (!curatedCdcDataFrame.isEmpty()) domainService
```

Comment:
Here is the entry point of the incremental domain refresh. The step takes the curated CDC dataframe as an input.